discovery/src/main/scala/akka/discovery/SimpleServiceDiscovery.scala

```diff
@@ -112,6 +112,32 @@ case object Lookup {
    * and protocol
    */
   def create(serviceName: String): Lookup = new Lookup(serviceName, None, None)
+
+  private val SrvQuery = """^_(.+?)\._(.+?)\.(.+?)$""".r
+
+  /**
+   * Create a service Lookup from a string with format:
+   * _portName._protocol.serviceName.
+   *
+   * If the passed string conforms with this format, a SRV Lookup is returned.
+   *
+   * The string is parsed and dismembered to build a Lookup as following:
+   * Lookup(serviceName).withPortName(portName).withProtocol(protocol)
+   *
+   * If the string doesn't not conform with the SRV format, a simple A/AAAA Lookup is returned.
+   */
+  def parseSrvString(srv: String): Lookup =
```

Comment:
I was thinking about the method name. Because we can pass a non-compliant String and still get a valid `Lookup` back, we should probably not call it `parse`. 

Falling to parse should be an error and that's not what we are doing here. 

Some suggestions:

```suggestion
  def fromString(str: String): Lookup =
```
or 
```suggestion
  def fromSrvString(srv: String): Lookup =
```

I prefer the first one because we can effectively pass any string and get a valid `Lookup`.